src/packlink.php

```diff
@@ -440,77 +416,257 @@ public function getOrderShippingCostExternal($params)
      *
      * @return string
      *
-     * @throws \Logeecom\Infrastructure\ORM\Exceptions\RepositoryNotRegisteredException
-     * @throws \Logeecom\Infrastructure\TaskExecution\Exceptions\QueueStorageUnavailableException
      * @throws \PrestaShopException
      */
     public function getContent()
     {
         \Packlink\PrestaShop\Classes\Bootstrap::init();
 
-        $output = '';
-
-        /** @var \Packlink\PrestaShop\Classes\BusinessLogicServices\ConfigurationService $configService */
-        $configService = \Logeecom\Infrastructure\ServiceRegister::getService(
-            \Packlink\BusinessLogic\Configuration::CLASS_NAME
-        );
-        $apiToken = $configService->getAuthorizationToken();
-        $loggedIn = false;
-
-        if (!$apiToken && Tools::isSubmit('api_key')) {
-            $loggedIn = $this->login();
-
-            if (!$loggedIn) {
-                $output .= $this->displayError(
-                    \Packlink\PrestaShop\Classes\Utility\TranslationUtility::__('API key was incorrect.')
-                );
-            }
-        }
-
         /** @var \Logeecom\Infrastructure\TaskExecution\Interfaces\TaskRunnerWakeup $wakeupService */
         $wakeupService = \Logeecom\Infrastructure\ServiceRegister::getService(
             \Logeecom\Infrastructure\TaskExecution\Interfaces\TaskRunnerWakeup::CLASS_NAME
         );
         $wakeupService->wakeup();
 
-        $fancyBoxUrl = _PS_BASE_URL_ . __PS_BASE_URI__ . 'js/jquery/plugins/fancybox/jquery.fancybox.js';
-        $fancyBoxUrl = str_replace(array('https:', 'http:'), '', $fancyBoxUrl);
+        $this->loadStyles();
+        $this->loadScripts();
+
+        \Packlink\BusinessLogic\Configuration::setCurrentLanguage($this->context->language->iso_code);
 
         $this->context->smarty->assign(array(
-            'fancyBoxPath' => $fancyBoxUrl,
+            'lang' => $this->getTranslations(),
+            'templates' => $this->getTemplates(),
+            'urls' => $this->getUrls(),
+            'baseResourcesUrl' => $this->getPathUri() . 'views/packlink',
+            'gridResizerScript' => $this->getPathUri() . 'views/packlink/js/GridResizerService.js?v=' . $this->version,
```

Comment:
It would make code much cleaner if all methods for retrieving needed data for the front end would be placed in another class in charge only for that part.

Reply:
This was not considered?